stock_available_to_promise_release/models/stock_move.py

```diff
@@ -392,9 +392,17 @@ def _release_split(self, remaining_qty):
         # `stock.move._search_picking_for_assignation`.
         if not self.picking_id.printed:
             self.picking_id.printed = True
-        new_move = self.browse(self._split(remaining_qty))
+
+        new_move_vals = self._split(remaining_qty)
+        if new_move_vals and isinstance(new_move_vals, list):
```

Comment:
Why checking the type of `new_move_vals` here?
Seems that `_split` always returns a list in standard: https://github.com/odoo/odoo/blob/32645a802db1c68f55d7de18d9f8ce22ee6b8a0c/addons/stock/models/stock_move.py#L1581-L1623

Reply:
Thanks @sebalix for the comment. When there is no available product in stock, the `_split` will be called with `self.product_qty`, https://github.com/odoo/odoo/blob/32645a802db1c68f55d7de18d9f8ce22ee6b8a0c/addons/stock/models/stock_move.py#L1594-L1595 will return empty list. Without that type checking, no move was returned by `create()` due to empty list, thus no new picking was assigned by `_assign_picking`. In turn, https://github.com/dzungtran89/wms/blob/bebdc2cea743515fb1c1711144e421010a664ae2/stock_available_to_promise_release/models/stock_move.py#L351 will add existing picking to backorder list. I believe that was not expected.